terraform/publisher/main.tf

```diff
@@ -0,0 +1,350 @@
+terraform {
+  backend "s3" {
+    bucket  = "govuk-terraform-test"
+    key     = "projects/app-publisher.tfstate"
+    region  = "eu-west-1"
+    encrypt = true
+  }
+}
+
+provider "aws" {
+  version = "~> 2.69"
+  region  = "eu-west-1"
+}
+
+#
+# Data
+#
+
+data "aws_vpc" "vpc" {
+  id = "vpc-9e62bcf8"
+}
+
+# TODO: Move publisher under GOV.UK Cluster
+# data "aws_ecs_cluster" "govuk" {
+#   cluster_name = "govuk"
+# }
+
+#
+# ECS Cluster, Service, Task
+#
+
+data "aws_iam_role" "task_execution_role" {
+  name = "fargate_task_execution_role"
+}
+
+resource "aws_ecs_cluster" "cluster" {
+  name               = var.service_name
+  capacity_providers = ["FARGATE"]
+}
+
+resource "aws_ecs_task_definition" "service" {
+  family                   = var.service_name
+  requires_compatibilities = ["FARGATE"]
+  container_definitions    = file("../task-definitions/publisher.json")
+  network_mode             = "awsvpc"
+  cpu                      = 2048
+  memory                   = 4096
```

Comment:
Should we try stick to the application sizes we have estimated within ESC Fargate cost calculator doc? It would be good to see if these estimates are sensible.